### PR TITLE
Fix router bugs, add HTTPS redirect feature.

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
Fix https://github.com/rancher/rio/issues/467 https://github.com/rancher/rio/issues/936

File: docs/router.md [hunk 1]
```diff
@@ -35,10 +35,18 @@ Creating a route with path-based matching supports: exact match: `foo`; prefix m
 $ rio route add $name/path to $target
 
 # to add prefix
-$ rio route add $name/path* to $target
+$ rio route add $name/path'*' to $target
```
Comment (cbron) [inline docs/router.md]: Looks like this was just outdated from before ? Didn't realize syntax changed.

File: modules/gloo/pkg/vsfactory/router.go [hunk 2]
```diff
@@ -226,10 +243,6 @@ func addRedirect(route riov1.RouteSpec, gatewayRoute *gatewayv1.Route) {
 		ra.RedirectAction.PathRewriteSpecifier = &gloov1.RedirectAction_PathRedirect{
 			PathRedirect: route.Redirect.Path,
 		}
-	} else if route.Redirect.Prefix != "" {
```
Comment (cbron) [inline modules/gloo/pkg/vsfactory/router.go]: Prefix is just handled via match now ?

File: docs/router.md (hunk 1 above)
Comment (StrongMonkey) [reply docs/router.md]: Doesn't work with bash cause you have to escape *.

File: modules/gloo/pkg/vsfactory/router.go (hunk 2 above)
Comment (StrongMonkey) [reply modules/gloo/pkg/vsfactory/router.go]: prefix is handled by route.Redirect.Path

**Pull request merged**